src/main/java/tictactoe/BoardConverter.java

```diff
@@ -11,38 +13,87 @@ public String toString(Marks[][] board) {
 
         for (Marks[] columns : board) {
             for (Marks mark : columns) {
-                str.append(createLine(idSpot, mark));
+                str.append(createSpot(idSpot, mark));
                 idSpot++;
             }
 
             if (++idLine != board.length) {
-                str.append("-----------\n");
+                str.append(repeatElement(totalCharBoardWidth(board), INTER_LINE));
+                str.append(NEW_LINE);
             }
         }
 
         return str.toString();
     }
 
-    private String createLine(int idSpot, Marks mark) {
-        return " " + markToDisplay(mark, idSpot) + rightEdgeSpot(idSpot);
+    private String createSpot(int idSpot, Marks mark) {
+        String centeredMark = center(correctMark(mark, idSpot), spotWidth(board));
+        return centeredMark + elementRightSpot(idSpot);
     }
 
-    private String rightEdgeSpot(int y) {
+    private final String CROSS = "X";
+    private final String ROUND = "O";
+    private String correctMark(Marks mark, int idSpot) {
+        if (mark == Marks.CROSS) {
+            return CROSS;
+        } else if (mark == Marks.ROUND) {
+            return ROUND;
+        } else {
+            return String.valueOf(idSpot);
+        }
+    }
+
+    private final String SPOT_BORDER = "|";
+    private String elementRightSpot(int y) {
         if (isBoardEdge(y)) {
-            return " \n";
+            return NEW_LINE;
         } else {
-            return " |";
+            return SPOT_BORDER;
         }
     }
 
-    private char markToDisplay(Marks mark, int idSpot) {
-        if (mark == Marks.CROSS) {
-            return 'X';
-        } else if (mark == Marks.ROUND) {
-            return 'O';
+    private int spotWidth(Marks[][] board) {
+        int idMaxWidth = String.valueOf(board.length * board.length - 1).length();
+        idMaxWidth += 2;
+        return idMaxWidth;
+    }
+
+    private int totalCharBoardWidth(Marks[][] board) {
+        int rightElements = board.length - 1;
+        return spotWidth(board) * board.length + rightElements;
+    }
+
+    private final String SPACE = " ";
+    private String center(String str, int totalWidth) {
+        int paddingWidth = calculatePaddingWidth(str, totalWidth);
+
+        if (totalWidthIsOdd(str, totalWidth)) {
+            return padding(paddingWidth) + str + padding(paddingWidth);
         } else {
-            return Character.forDigit(idSpot, 10);
+            return padding(paddingWidth) + SPACE + str + padding(paddingWidth);
+        }
+    }
+
+    private int calculatePaddingWidth(String str, int width) {
+        return (width - str.length()) / 2;
+    }
+
+    private boolean totalWidthIsOdd(String str, int width) {
+        return (str.length() + width) % 2 == 0;
+    }
+
+    private String padding(int width) {
+        return repeatElement(width, SPACE);
+    }
+
+    private String repeatElement(int width, String str) {
+        StringBuilder spaces = new StringBuilder();
```

Comment:
Mentioned earlier, but for completeness' sake `StringBuilder` is deprecated.

Main thing I wanted to point out is that the name `spaces` is lying here. It's no spaces that get repeated here but whatever has been passed by `str`.